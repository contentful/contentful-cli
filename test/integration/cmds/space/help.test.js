const nixt = require('nixt')
const { join } = require('path')

const bin = join(__dirname, './../../../../', 'bin')

const app = () => {
  return nixt({ newlines: true })
    .cwd(bin)
    .base('./contentful.js ')
    .clone()
}

test('should print help message', done => {
  app()
    .run('space --help')
    .code(0)
    .expect(result => {
      const resultText = result.stdout.trim()
      expect(resultText).toMatchSnapshot('help data is incorrect')
    })
    .end(done)
})

test('should print help message when no command provided', done => {
  app()
    .run('space')
    .code(1)
    .expect(result => {
      const resultText = result.stderr.trim()
      expect(resultText).toMatchSnapshot(
        'wrong response in case of no command provided'
      )
    })
    .end(done)
})

test('should print help message on shortcut', done => {
  app()
    .run('space -h')
    .code(0)
    .expect(result => {
      const resultText = result.stdout.trim()
      expect(resultText).toMatchSnapshot('help data is incorrect')
    })
    .end(done)
})
