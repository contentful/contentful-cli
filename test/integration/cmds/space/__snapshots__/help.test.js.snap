// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should print help message on shortcut: help data is incorrect 1`] = `
"Commands:
  space accesstoken         Manage and list your delivery access tokens
                                                                   [aliases: at]
  space environment-alias   Manage and list environment aliases
  space create              Create a space
  space delete              Deletes a space
  space environment         Manage and list environments
  space export              export a space data to a json file
  space generate            Generate code for different tasks       [aliases: g]
  space import              import a space
  space list                List your spaces                       [aliases: ls]
  space migration           Parses and runs a migration script on a Contentful
                            space
  space seed                Seed a content model and content based on given
                            templates
  space use                 Sets the default space which every command will use
                            when the --space-id option is skipped.  [aliases: u]

Options:
  -h, --help  Show help                                                [boolean]"
`;

exports[`should print help message when no command provided: wrong response in case of no command provided 1`] = `
"Commands:
  space accesstoken         Manage and list your delivery access tokens
                                                                   [aliases: at]
  space environment-alias   Manage and list environment aliases
  space create              Create a space
  space delete              Deletes a space
  space environment         Manage and list environments
  space export              export a space data to a json file
  space generate            Generate code for different tasks       [aliases: g]
  space import              import a space
  space list                List your spaces                       [aliases: ls]
  space migration           Parses and runs a migration script on a Contentful
                            space
  space seed                Seed a content model and content based on given
                            templates
  space use                 Sets the default space which every command will use
                            when the --space-id option is skipped.  [aliases: u]

Options:
  -h, --help  Show help                                                [boolean]
Please specify a sub command."
`;

exports[`should print help message: help data is incorrect 1`] = `
"Commands:
  space accesstoken         Manage and list your delivery access tokens
                                                                   [aliases: at]
  space environment-alias   Manage and list environment aliases
  space create              Create a space
  space delete              Deletes a space
  space environment         Manage and list environments
  space export              export a space data to a json file
  space generate            Generate code for different tasks       [aliases: g]
  space import              import a space
  space list                List your spaces                       [aliases: ls]
  space migration           Parses and runs a migration script on a Contentful
                            space
  space seed                Seed a content model and content based on given
                            templates
  space use                 Sets the default space which every command will use
                            when the --space-id option is skipped.  [aliases: u]

Options:
  -h, --help  Show help                                                [boolean]"
`;
